src/Promise.fs

```diff
@@ -454,6 +454,142 @@ let Parallel (pr: seq<JS.Promise<'T>>): JS.Promise<'T[]> = jsNative
+///     match success |> Promise.settledResult with
+///     | Ok value -> value |> equal 1
+///     | _ -> failwith "Unreachable result"
+///     
+///     match rejection |> Promise.settledResult with
+///     | Error ex -> ex.Message |> equal "I Failed"
+///     | _ -> failwith "Unreachable result"
+/// )
+/// </code>
+/// </example>
+let settledResult (p: SettledValue<'T>) =
+    match p with
+    | Success value -> Ok value
+    | Rejection err -> Error err
+
+/// <summary>
+/// The Promise.allSettled function returns a promise that resolves after all of the given promises have either
+/// fulfilled or rejected, with an array of objects that each describes the outcome of each promise.
+/// It is typically used when you have multiple asynchronous tasks that are not dependent on one another to complete successfully,
+/// or you'd always like to know the result of each promise.
+/// In comparison, the Promise returned by Promise.all may be more appropriate if the tasks are dependent
+/// on each other / if you'd like to immediately reject upon any of them rejecting.
+/// </summary>
+/// <param name="pr">A list of promise to wait for</param>
+/// <typeparam name="'T"></typeparam>
+/// <example>
+/// <code lang="fsharp">
+/// let success =
+///     promise {
+///         do! Promise.sleep 100
+///         return 1
+///     }
+/// let rejection = Promise.reject (exn "I Failed")
+/// 
+/// Promise.allSettled [success; rejection]
+/// |> Promise.map(fun values ->
+///     let success = values.[0]
+///     let rejection = values.[1]
+///     success.value.Value |> equal 1
+///     rejection.reason.Value.Message |> equal "I Failed"
+/// )
+/// </code>
+/// </example>
+[<Emit("Promise.allSettled($0)")>]
+let allSettled (pr: seq<JS.Promise<'T>>): JS.Promise<SettledValue<'T>[]> = jsNative
+
+/// <summary>
+/// Promise.any takes an iterable of Promise objects.
+/// It returns a single promise that resolves as soon as
+/// any of the promises in the iterable fulfills,
+/// with the value of the fulfilled promise. If no promises in the iterable fulfill
+/// (if all of the given promises are rejected),
+/// then the returned promise is rejected with an AggregateError,
+/// a new subclass of Error that groups together individual errors.
+/// </summary>
+/// <param name="pr">A list of promise to wait for</param>
+/// <typeparam name="'T"></typeparam>
+/// <example>
+/// <code lang="fsharp">
+/// let success() =
+///     promise {
+///         do! Promise.sleep 100
+///         return 1
+///     }
+/// let rejection() = Promise.reject (exn "I Failed")
+/// let rejection2() =
+///     promise {
+///         do! Promise.sleep 50
+///         return! Promise.reject (exn "I Failed")
+///     }
+/// 
+/// Promise.any [rejection(); rejection2(); success()]
+/// |> Promise.map (fun result ->
+///     result |> equal 1
+/// )
+/// </code>
+/// </example>
+[<Emit("Promise.any($0)")>]
+let any (pr: seq<JS.Promise<'T>>): JS.Promise<'T> = jsNative
```

Comment:
I think I had a similar helper in one of my projects. The tricky thing is how to deal with promises returning different types. My helper was mainly to use timeouts so I was wrapping `Promise.race` to use `Choice`, though this means we would need `race2`, `race3`, `race4`... (same for `any`). This also makes me think if we need an untyped `Promise` 🤔

Reply:
Would erased unions could ease those cases?

if that is the case I could add some docs about it

Reply:
The erased unions were specifically designed for the bindings. They can be used sometimes in your own Fable code but you need to be careful. Here I don't think they're a good replacement for Choice because pattern matching becomes type testing and this is not reliable many times (e.g. when you have two cases with the same type or when some results are just pojos and not typed in runtime).